environments/spring.py

```diff
@@ -0,0 +1,114 @@
+from matplotlib import pyplot as plt, animation
+import numpy as np
+
+from environments import Environment
+
+
+class Spring(Environment):
+
+    """Spring System
+
+    Equations of movement are:
+
+        x'' = -(k/m)*x
+
+    """
+
+    def __init__(self, mass, elastic_cst, p=None, q=None):
+        """Contructor for spring system
+
+        Args:
+            mass (float): Spring mass
+            elastic_cst (float): Spring elastic constant
+            p ([float], optional): Generalized momentum in 1-D space: Linear momentum (kg*m/s). Defaults to None
+            q ([float], optional): Generalized position in 1-D space: Position (m). Defaults to None
+        """
+        self.mass = mass
+        self.elastic_cst = elastic_cst
+        super().__init__(p, q)
```

Comment:
Its better to do `(q=q, p=p)` to make sure we don't do mistakes with the position and momentum